ambry-api/src/main/java/com.github.ambry/rest/RequestPath.java

```diff
@@ -91,9 +95,18 @@ public static RequestPath parse(String path, Map<String, Object> args, List<Stri
     // test if the last segment is a sub-resource
     RestUtils.SubResource subResource = null;
     int lastSlashOffset = path.lastIndexOf('/');
+    long blobSegmentNum = NO_BLOB_SEGMENT_IDX_SPECIFIED;
+    System.out.println(Arrays.asList(path.split("/")));
     if (lastSlashOffset > offset) {
       try {
-        subResource = RestUtils.SubResource.valueOf(path.substring(lastSlashOffset + 1));
+        String[] fields = path.split("/");
+        if (RestUtils.SubResource.Segment.toString().equals(fields[fields.length - 2])) {
```

Comment:
This is a mouthful, suggest importing SubResource statically.
Maybe do:
```
// top of class
private static final String SEGMENT = SubResource.Segment.toString();
// here
String maybeSegmentField = fields[fields.length - 2];
if (SEGMENT.equals(maybeSegmentField)) { // or equalsIgnoreCase
    subResource = SubResource.Segment;
    ...
}
```

Reply:
done